### ISSUE TITLE
Several implementation details with the paper and code.

### ISSUE THREAD

**Issue by wutong16** [leokarlin/LaSO#5]
Hi, thanks for your great work and it's really impressing!
I have some problem when reproducing the results and I'd like to verify some implementation details according to your paper and the code~

- I got confused when reading the following descriptions of your paper in section4:
>  We filtered the COCO train set leaving only images that did not contain any of the 16 unseen category labels and used this filtered set to train our feature extractor backbone (InceptionV3) and the LaSO models.
> For the 16 unseen categories, the 16-way classifier, used for the evaluation, was pre-trained on the images of the COCO training set containing instances of these 16 categories.

So, when training the backbone and the  LaSO network, are you using only part of COCO training set with images containing **no 16-unseen instances** or the whole COCO training set and just **masking the 16-unseen labels** ?

- When evaluating the results, for the 64-way or 16-way classifier, we extract images from the COCO val2014 set that **contain at least one label in the 64/16 label set** and evaluate on the extracted image set, is that right? 

- As for the results(only considering the **original feature vectors**): the 75 and 79 mAP in Table 1 is really high, may I ask what the baseline(inception3-pooling-FC, training on filtered COCO dataset with 64-seen classes, basic augmentation without LaSO) results are?    And the same problem appears when I'm trying out the baseline B(basic data augmentation with k samples for each novel class) for few shot testing, I can only get 34 mAP for 5-shot testing, where it should be 50+ in your paper. Here I've tried the locked backbone of InceptionV3 with SGD init_lr=0.001 or Adam init_lr=0.001.

I think there must be something wrong with my coding and your reply will be highly appreciated!

**Comment by alfassy** [comment]
> So, when training the backbone and the LaSO network, are you using only part of COCO training set with images containing no 16-unseen instances or the whole COCO training set and just masking the 16-unseen labels ?
The first option.

> When evaluating the results, for the 64-way or 16-way classifier, we extract images from the COCO val2014 set that contain at least one label in the 64/16 label set and evaluate on the extracted image set, is that right?
Yes.

As for the last question, the 75 and 79 can be received using the coco inception v3 model which we've published here, also during the training of LaSO, we also print the mAP on the real original vectors, so you can just run the training script for 1 epoch and get the 75. The paperDiscriminator model we've published can reproduce the 79 for you when using our augmentation code.
As for baseline B2, I am not sure which code you've used, but you can take our augmentation code, comment out the LaSO augmentation in the training loop, add basic augmentations and get approximately these results, notice that you might have to change the learning rate.

Sorry I couldn't help more.

**Comment by wutong16** [comment]
@alfassy Thanks so much for your clear and timely response and it's of great help! I'll give it a try!

**Comment by wutong16** [comment]
Hi, sorry to bother you again, but I'm not sure what the following statement means in the paper:

>For the 16 unseen categories, the 16-way classifier, used for the evaluation, was pre-trained on the images of the COCO training set containing instances of these 16 categories.

So apart from the 1-shot and 5-shot testing time, where are the images that contain 16-unseen instances used in the pre-training stage?

Thanks very much and looking forward to your reply!

**Comment by alfassy** [comment]
Hi, the 16 unseen instances are also used in order to train the paperDiscriminator, with this discriminator we can check the classification accuracy of the LaSO vectors on the 16 unseen vectors.

**Comment by wutong16** [comment]
I see~ Thanks very much!
